flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/LocalChangelogRegistryImpl.java

```diff
@@ -87,8 +88,8 @@ public void discardUpToCheckpoint(long upTo) {
     public void prune(long checkpointID) {
         Set<StreamStateHandle> handles =
                 handleToLastUsedCheckpointID.values().stream()
-                        .filter(tuple -> tuple.f1 == checkpointID)
-                        .map(tuple -> tuple.f0)
+                        .filter(entry -> entry.createdCheckpointId == checkpointID)
```

Comment:
Agreed, the pre-emptive upload files would not be re-upload, so those aborted files may be referenced by subsequent checkpoints.  But in case of many subsequent aborted checkpoints, local state accumulates, all aborted local state will not be deleted until the next completed materialization. 

Considering that the priority of correctness is higher than that of space amplification, I first deleted this method.